lizard_auth_client/middleware.py

```diff
@@ -60,3 +62,27 @@ def process_request(self, request):
             path = request.path_info.strip('/')
             if not string_has_any_prefix(exempt_urls, path):
                 return HttpResponseRedirect(settings.LOGIN_URL)
+
+
+class AttemptAutoLoginMiddleware(object):
+    """Apply the attempt_auto_login decorator on every view function.
+
+    Note: this middleware doesn't work together with LoginRequiredMiddleware
+    because it intercepts the requests before they can be handled here.
+    """
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        assert hasattr(request, 'user'), '''The AttemptAutoLoginMiddleware
+        requires authentication middleware to be installed. Edit your
+        MIDDLEWARE_CLASSES setting to insert
+        'django.contrib.auth.middlware.AuthenticationMiddleware'.
+        If that doesn't work, ensure your TEMPLATE_CONTEXT_PROCESSORS
+        setting includes 'django.core.context_processors.auth'.'''
+
+        path = request.path_info.strip('/')
+        if string_has_any_prefix(exempt_urls, path):
```

Comment:
Ik vind het onduidelijk lezen dat `exempt_urls` niet UPPERCASE is, dat zou je voor zo'n constante (dat het eigenlijk is ondanks dat er wat aan gerekend wordt) verwachten.